### PR TITLE
Canvas: disable delete button when there is only one page

### LINKED ISSUE #783: Delete icon should be disabled when story only contains a single page
## Bug Description

Having a one-page story and deleting its only page has no effect. 

## Expected Behaviour

We can go multiple ways about this. Pressing the button should either:
- Empty out the page and reset its background/properties
- Actually delete the page and show an empty state
- Not delete the page and show a snackbar ("Can't delete the only page left")

## Steps to Reproduce

- Press the delete page button on a story that only has one page

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by g-sqiao**
## Context

Snip20210813_5
Snip20210813_6

### QA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [ ] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [ ] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [ ] I have tested this code to the best of my abilities
- [ ] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [ ] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [ ] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [ ] I have added documentation where necessary
- [ ] I have added a matching `Type: XYZ` label to the PR

---

Fixes #783

### CODE REVIEW

File: packages/story-editor/src/components/canvas/pagemenu/pageMenu.js [hunk 1]
```diff
@@ -63,17 +63,19 @@ function PageMenu() {
...
   } = useStory(
     ({
-      state: { currentPageNumber, currentPage },
+      state: { currentPageNumber, currentPage, pages },
       actions: { deleteCurrentPage, addPage },
     }) => {
       return {
         currentPageNumber,
         currentPage,
+        pages,
```
Comment (dreamofabear) [inline packages/story-editor/src/components/canvas/pagemenu/pageMenu.js]:
```suggestion
        numberOfPages: pages.length,
```

File: packages/story-editor/src/components/canvas/pagemenu/pageMenu.js [hunk 2]
```diff
@@ -63,17 +63,19 @@ function PageMenu() {
   const {
     currentPageNumber,
     currentPage,
+    pages,
```
Comment (dreamofabear) [inline packages/story-editor/src/components/canvas/pagemenu/pageMenu.js]:
```suggestion
    numberOfPages,
```

File: packages/story-editor/src/components/canvas/pagemenu/pageMenu.js [hunk 3]
```diff
@@ -101,6 +103,7 @@ function PageMenu() {
   );
 
   const isWidePage = pageWidth > 280;
+  const disableDeleteButton = pages && pages.length === 1;
```
Comment (dreamofabear) [inline packages/story-editor/src/components/canvas/pagemenu/pageMenu.js]:
```suggestion
  const disableDeleteButton = numberOfPages <= 1;
```

**Review by dreamofabear — APPROVED** [review]
Just a minor suggestion to make the context selector more specific.

Optional: Looks like there's no `pagemenu/test/` but it'd be nice to add one as a follow-up. You can follow the example from other tests e.g. `pageAttachment/test/`.

**Review by swissspidy — APPROVED** [review]
LGTM 👍 

As Will said, adding some unit tests as a follow-up would be nice.

**Pull request merged**